pkg/splunk/enterprise/util.go

```diff
@@ -1044,11 +1245,20 @@ func initAndCheckAppInfoStatus(client splcommon.ControllerClient, cr splcommon.M
 				return err
 			}
 
+			// TODO: gaurav, this needs to be removed eventually in favor of downloading apps locally
 			_, _, err = ApplyAppListingConfigMap(client, cr, appFrameworkConf, appStatusContext.AppsSrcDeployStatus)
 			if err != nil {
 				return err
 			}
 
+			// TODO: gaurav, this will eventually go outside this if condition,
```

Comment:
So does this mean that we will handle all app downloads/installs in a single iteration of the reconcile loop?  Do we wait on all worker threads to finish before this call returns?  Will this block servicing other reconciles?  I'm fine with the caveat but we should probably make that explicitly know in this comment.

Reply:
As of right now, yes we are doing everything in one reconcile and will wait until all the downloads are finished. @sgontla is going to introduce the yield logic through which we can continue the work across reconciles.